src/Components/Projects/Projects.tsx

```diff
@@ -138,25 +151,54 @@ const Projects: React.FC = () => {
         );
       });
   };
-
+  // **********search by proj name***********************
+  const getProjectTitleValue = (e: MouseEvent) => {
+   
+    getAllProjectsList(1, e.target.value);
+    setSearchString(e.target.value);
+   
+  };
+  // *****************************************************
   useEffect(() => {
-    getAllProjectsList();
+      getAllProjectsList(1,searchString);
```

Comment:
Consider the following refined version:

This function is invoked only upon the component rendering when `searchString` is an empty string. To avoid potential confusion for other developers reviewing the code, it's advisable to directly pass an empty string ("") or null instead of explicitly writing `searchString`. This eliminates the need to include `searchString` in the dependency array, streamlining the code for better clarity.

Alternatively, you can express the function call as `getAllProjectsList();`. However, for improved flexibility and clarity, it is recommended to make function parameters optional or provide default values. For instance:

```typescript
getAllProjectsList = (pageNo: number = 5, title: string = "") => { ... }
```

Personally, I find the second approach preferable, as it not only makes the function more versatile but also enhances readability by explicitly defining default values for parameters.

Reply:
Here I made a mistake by defaulting the page number to 5, I thought it was the Page Size.
Sorry for the confusion.